tiled/client/context.py

```diff
@@ -247,27 +247,49 @@ def get_content(self, path, accept=None, timeout=UNSET, stream=False, **kwargs):
                 return blosc.decompress(response.content)
             return response.content
         # If we get this far, we have an online client and a cache.
-        reservation = self._cache.get_reservation(url)
+        # Parse Cache-Control header directives.
+        cache_control = {
+            directive.lstrip(" ")
+            for directive in request.headers.get("Cache-Control", "").split(",")
+        }
+        if "no-cache" in cache_control:
+            reservation = None
+        else:
+            reservation = self._cache.get_reservation(url)
         try:
             if reservation is not None:
+                if not reservation.is_stale():
+                    # Short-circuit. Do not even bother consulting the server.
+                    return reservation.load_content()
```

Comment:
either do multiple returns or single return is this function, doing a direct return here and then setting up `content` to be the return value below was confusing.

Reply:
Done